packages/flutter/test/painting/text_painter_test.dart

```diff
@@ -633,5 +633,5 @@ void main() {
     );
     expect(caretOffset.dx, closeTo(0.0, 0.0001));
     expect(caretOffset.dy, closeTo(0.0, 0.0001));
-  });
+  }, tags: 'web_unimplemented');
```

Comment:
why is this not implemented?

Reply:
We should treat `web_unimplemented` as bugs. Frequently it would be some feature that we have not synced with yet.

PS: this test could be broken up into multiple smaller tests; would make it easier to find the incompatibility.

Reply:
/cc @mdebbar (perhaps you can spot the issue right away)

Reply:
It's also possible that some of these are due to me hacking out the retained mode API for compatibility, or issues that have otherwise been fixed since

Reply:
This test is skipped in flutter_web, but I tried running it and it's passing (we probably forgot to unskip it when the underlying failure was fixed). So the failure here must be due to some recent-ish changes in Flutter that we aren't handling properly on the web.